src/raw_node.rs

```diff
@@ -349,94 +385,299 @@ impl<T: Storage> RawNode<T> {
         Err(Error::StepPeerNotFound)
     }
 
-    /// Given an index, creates a new Ready value from that index.
-    pub fn ready_since(&mut self, applied_idx: u64) -> Ready {
-        Ready::new(
-            &mut self.raft,
-            &self.prev_ss,
-            &self.prev_hs,
-            Some(applied_idx),
-        )
-    }
-
     /// Ready returns the current point-in-time state of this RawNode.
     pub fn ready(&mut self) -> Ready {
-        Ready::new(&mut self.raft, &self.prev_ss, &self.prev_hs, None)
+        let raft = &mut self.raft;
+
+        self.max_number += 1;
+        let mut rd = Ready {
+            number: self.max_number,
+            ..Default::default()
+        };
+        let mut rd_record = ReadyRecord {
+            number: self.max_number,
+            ..Default::default()
+        };
+
+        // If there is a pending snapshot, do not get the whole Ready.
+        if self.pending_snapshot {
```

Comment:
Why not check it at the beginning of the function?

Reply:
Because the `Ready` and its corresponding  `ReadyRecord` should be created.

Reply:
Why not return `None` when `pending_snapshot` is true?

Reply:
Do you mean `ready` return `Option<Ready>`? I think it's not necessary because users will call `has_ready` in general cases.